state.go

```diff
@@ -356,24 +356,52 @@ func (i *InitState) Handle(ctx *StateContext, c Controller) (State, bool) {
 
 	log.Infof("handling loaded state: %s", sd.Name)
 
-	// chack last known state
-	switch sd.Name {
-	// update process was finished; check what is the status of update
-	case MenderStateReboot:
+	if !committedPartition(c) {
+		// only valid entrypoint into the uncommitted partition is reboot_leave
+		if sd.Name != MenderStateReboot {
+			// entered the uncommitted partition without finishing the whole update-path
+			// on the committed partition, therefore reboot back into the committed partition
+			// and error
+			if err := c.Reboot(); err != nil {
+				return NewUpdateErrorState(NewFatalError(errors.Errorf("failed to reboot device: %v", err)), sd.UpdateInfo), false
+			}
+			// should never happen
+			return doneState, false
+		}
 		return NewAfterRebootState(sd.UpdateInfo), false
+	}
+
+	// check last known state
+	switch sd.Name {
 
 	case MenderStateRollbackReboot:
 		return NewAfterRollbackRebootState(sd.UpdateInfo), false
 
-	// this should not happen
+	// Rerun commit-leave
+	case MenderStateUpdateCommit:
```

Comment:
Do you want to force the update_commit -> init transition here? Why can't we simply skip setting the update_commit state?